pkg/registry/populator_test.go

```diff
@@ -868,3 +871,250 @@ func TestDeprecateBundle(t *testing.T) {
+						"beta",
+						"alpha",
+						"stable",
+					},
+					"prometheus": []string{
+						"preview",
+						"stable",
+						"alpha",
+					},
+				},
+				remainingDefaultChannels: map[string]string{
+					"etcd":       "alpha",
+					"prometheus": "preview",
+				},
+			},
+		},
+		{
+			description: "OverwriteBundle/DefaultChannelChange",
+			args: args{
+				firstAdd: getBundleRefs([]string{"prometheus.0.14.0", "prometheus.0.15.0"}),
+				secondAdd: map[image.Reference]string{
+					image.SimpleReference("quay.io/test/etcd.0.9.0"):            "../../bundles/etcd.0.9.0",
+					image.SimpleReference("quay.io/test/etcd.0.9.2"):            "../../bundles/etcd.0.9.2",
+					image.SimpleReference("quay.io/test/new-prometheus.0.15.0"): "testdata/overwrite/prometheus.0.15.0",
+				},
+			},
+			expected: expected{
+				err: nil,
+				remainingBundles: []string{
+					"quay.io/test/etcd.0.9.0/alpha",
+					"quay.io/test/etcd.0.9.0/beta",
+					"quay.io/test/etcd.0.9.0/stable",
+					"quay.io/test/etcd.0.9.2/alpha",
+					"quay.io/test/etcd.0.9.2/stable",
+					"quay.io/test/prometheus.0.14.0/preview",
+					"quay.io/test/prometheus.0.14.0/stable",
+					"quay.io/test/new-prometheus.0.15.0/preview",
+					"quay.io/test/new-prometheus.0.15.0/stable",
+				},
+				remainingPkgChannels: pkgChannel{
+					"etcd": []string{
+						"alpha",
+						"beta",
+						"stable",
+					},
+					"prometheus": []string{
+						"preview",
+						"stable",
+					},
+				},
+				remainingDefaultChannels: map[string]string{
+					"etcd":       "alpha",
+					"prometheus": "stable",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			logrus.SetLevel(logrus.DebugLevel)
+			db, cleanup := CreateTestDb(t)
+			defer cleanup()
+			store, err := sqlite.NewSQLLiteLoader(db)
+			require.NoError(t, err)
+			require.NoError(t, store.Migrate(context.TODO()))
+
+			query := sqlite.NewSQLLiteQuerierFromDb(db)
+
+			graphLoader, err := sqlite.NewSQLGraphLoaderFromDB(db)
+			require.NoError(t, err)
+
+			populate := func(bundles map[image.Reference]string) error {
+				return registry.NewDirectoryPopulator(
+					store,
+					graphLoader,
+					query,
+					bundles,
+					true).Populate(registry.ReplacesMode)
+			}
+			require.NoError(t, populate(tt.args.firstAdd))
+			require.NoError(t, populate(tt.args.secondAdd))
+
+			// Ensure remaining bundlePaths in db match
+			bundles, err := query.ListBundles(context.Background())
+			require.NoError(t, err)
+			var bundlePaths []string
+			for _, bundle := range bundles {
+				bundlePaths = append(bundlePaths, strings.Join([]string{bundle.BundlePath, bundle.ChannelName}, "/"))
+			}
+			fmt.Println("remaining", bundlePaths)
+			require.ElementsMatch(t, tt.expected.remainingBundles, bundlePaths)
+
+			// Ensure remaining channels and default channel match
+			packages, err := query.ListPackages(context.Background())
+			require.NoError(t, err)
+
+			for _, pkg := range packages {
```

Comment:
this doesn't catch when a package in `remainingPkgChannels` doesn't exist in the database